Add filter families active and inactive

scara82 opened the pull request:
This PR:
- Add buttons in families#index to show the active/notActive/all families;
- Adds feature specs for the 3 scenarios;
- Create spec/features/families/* and spec/features/schools/* and spec/features/trips/* for a better organisations of the specs.

Trello Card: [https://trello.com/c/fbGbXTaf](url)
<img width="1247" alt="screen shot 2018-03-03 at 5 47 50 pm" src="https://user-images.githubusercontent.com/25237403/36931507-0dd65ae4-1f0b-11e8-86e7-7fc954eac9c0.png">

Commit:
Remove trailing space

Commit:
Add active and not_active scopes for families

Commit:
Add sub-header and the partials in the index page

Commit:
Add javascript for show/hide families list partials

Commit:
Add javascript for show/hide families list partials

Commit:
Add javascript for show/hide families list partials

Commit:
Add javascript for show/hide families list partials

Commit:
Create specific folders for each Controller into spec/features and mov the files

Commit:
Create specific folders for each Controller into spec/features and mov the files

Commit:
Create specific folders for each Controller into spec/features and mov the files

Commit:
Create specific folders for each Controller into spec/features and mov the files

Commit:
Add features specs for happy path when admin click on notActive/Active/All in the index page and wants to see the different families lists

Commit:
Add features specs for happy path when admin click on notActive/Active/All in the index page and wants to see the different families lists

Commit:
Add features specs for happy path when admin click on notActive/Active/All in the index page and wants to see the different families lists

Commit:
Add features specs for happy path when admin click on notActive/Active/All in the index page and wants to see the different families lists

Commit:
Change translations

Commit:
Change translations

Commit:
Change translations

apalski commented on app/assets/stylesheets/components/_sub_header.scss:
```diff
@@ -33,3 +33,15 @@
+
+#all_families {
+  display: none;
+}
```
Thought about suggesting you alphabetise these and the functions above but then I thought they are probably grouped logically from how they connect.

scara82 replied:
Yeah, Elle said that the classes don't need to be in order also bc sometimes they need to have a certain order so I grouped by logic! 😉

apalski commented on spec/features/families/admin_views_all_families_spec.rb:
```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+feature "Admin can see all the families when in families#index" do
+  context "when click on the 'All' button" do
```
I think this needs to be scenario not context?

scara82 replied:
Oh, thanks, I was getting a little bit confused trying to figure out what to use!

scara82 replied:
Done with fixup in `4d01de2`

apalski commented:
Just a couple of comments, otherwise LGTM! Very cool use of JS too!

elle commented on app/assets/stylesheets/components/_sub_header.scss:
```diff
@@ -33,3 +33,15 @@
+
+#all_families {
+  display: none;
+}
```
How about merging these two rules together? i.e.
```
#all_families, #not_active_families {
  display: none;
}
```

scara82 replied:
Done with fixup in `9dfcb0f`

elle commented on app/controllers/families_controller.rb:
```diff
@@ -1,6 +1,8 @@
   def index
     @families = Family.all
+    @active_families = Family.active
+    @not_active_families = Family.not_active
```
A suggestion to change the wording to be `.inactive`

scara82 replied:
Done in sha: `e43fa21`

elle commented on app/models/family.rb:
```diff
@@ -1,3 +1,6 @@
   validates_presence_of :last_name, :capability
+
+  scope :active, -> { where("active = true") }
+  scope :not_active, -> { where("active = false") }
```
Use `def self.active` syntax for scopes as suggested by the thoughtbot style guide:
> Use def self.method, not the scope :method DSL.
Ref: https://github.com/thoughtbot/guides/tree/master/style/rails

elle replied:
Also, this can be written to: `where(active: true)`

scara82 replied:
Done in fixup `9deebe3`

elle commented on app/views/families/index.html.erb:
```diff
@@ -1,6 +1,25 @@
+<%= render "sub_header" %>
+<div id="active_families">
+  <p class="title"><%= t(".active_families") %></p>
+  <%= link_to t(".add_family"), new_family_path, class: "link_button, button" %>
```
Does this link belong inside of this `div`?

scara82 replied:
Yes, I did it in purpose because I want the `add family` button only in the page of the active families. ☺️

elle commented on config/locales/en.yml:
```diff
@@ -47,13 +47,15 @@ en:
-      submit: "Add"
       destroy: "Delete"
       edit: "Edit"
+      active_families: "Families Active"
```
It reads slightly better as "Active families"

scara82 replied:
Dona in sha `3e97938`

elle commented on config/locales/en.yml:
```diff
@@ -47,13 +47,15 @@ en:
       destroy: "Delete"
       edit: "Edit"
+      active_families: "Families Active"
+      not_active_families: "Families Not Active"
```
Similar here: it reads better as "Inactive families"

scara82 replied:
Done in sha `3e97938`

elle commented on config/locales/en.yml:
```diff
@@ -47,13 +47,15 @@ en:
+      all_families: "All families"
+      button_active_families: "Active"
+      button_not_active_families: "Not Active"
+      button_all_families: "All"
```
What do you think about nesting the buttons under a `buttons` key? i.e.
```
families:
  index:
    buttons:
      active:
      inactive:
      all:
```

scara82 replied:
Done in sha `3e97938`

elle commented on spec/features/families/admin_views_all_active_families_spec.rb:
```diff
@@ -0,0 +1,9 @@
+  scenario "successfully" do
+    visit families_path
+
+    expect(page).to have_text(I18n.t("families.index.active_families"))
```
This spec is missing actually seeing a family on the page.
I would add an active family and an inactive family, and check which one is displayed on the page

scara82 replied:
I created traits for all_true_families and all_false_families to test it properly because I have true/false also for bus_drivers and prof. I modified all 3 specs to use it, done in sha `b6c4d17`

elle commented on spec/features/families/admin_views_all_families_spec.rb:
```diff
@@ -0,0 +1,19 @@
+  context "when click on the 'All' button" do
+    scenario "successfully" do
+      create(:family, active: false, last_name: "Scara")
+      create(:family, active: true, last_name: "Liso")
```
Consider adding a trait in the family factory, so that you can just use:
`create(:family, :active)`

scara82 replied:
Modified in sha `b6c4d17`

elle commented on spec/features/families/admin_views_all_families_spec.rb:
```diff
@@ -0,0 +1,19 @@
+      expect(page).to have_text(I18n.t("families.index.all_families"))
+      expect(page).to have_text("false")
+      expect(page).to have_text("Scara")
+      expect(page).to have_text("true")
```
I wouldn't care so much about the assertions on lines 13 and 15, and just leave the assertions on lines 14 and 16

scara82 replied:
Modified in sha `b6c4d17`

elle commented on spec/features/families/admin_views_all_not_families_spec.rb:
```diff
@@ -0,0 +1,16 @@
+feature "Admin can see all the not active families when in families#index" do
+  context "when click on the 'Not Active' button" do
+    scenario "successfully" do
+      create(:family, active: false, last_name: "Scara")
```
Can we add an active family and check that we can't see it on the page?

elle commented on spec/features/families/admin_views_all_not_families_spec.rb:
```diff
@@ -0,0 +1,16 @@
+      find("#not_active_families_button").click
+
+      expect(page).to have_text(I18n.t("families.index.not_active_families"))
+      expect(page).to have_text("false")
```
Again, I would change this to check that the active family is not displayed on the page

scara82 replied:
Modified in sha `b6c4d17`

elle commented on app/assets/javascripts/families.js:
```diff
@@ -0,0 +1,17 @@
+  document.getElementById("all_families").style.display = "block";
+  document.getElementById("not_active_families").style.display = "none";
+  document.getElementById("active_families").style.display = "none";
+}
```
A suggestion to rewrite this to use a function to hide all, and then just turn one on.
Also, to use css classes for the hiding and showing by adding/removing a class. For example:

```
const buttons = document.querySelectorAll(".button-families");

function showFamilies(id_name) {
  buttons.forEach(button => button.classList.remove("active"));
  document.querySelector(id_name).classList.add("active");
}
```

elle replied:
Actually, you can build it to be generic enough, so it will also work with Trips. Happy to help with this if you wish

Commit:
Rename not_active_families into inactive_families

Commit:
Rename not_active_families into inactive_families

Commit:
Rename not_active_families into inactive_families

Commit:
Rename not_active_families into inactive_families

Commit:
Create a families.index.button group in transaltions and replace that in the specs and view

Commit:
Create a families.index.button group in transaltions and replace that in the specs and view

Commit:
Create a families.index.button group in transaltions and replace that in the specs and view

Commit:
Create a families.index.button group in transaltions and replace that in the specs and view

Commit:
Load JS after the DOM have been loaded so we can access DOM elements

Commit:
Load JS after the DOM have been loaded so we can access DOM elements

Commit:
Load JS after the DOM have been loaded so we can access DOM elements

Commit:
Clean up how tabs are being displayed

- Dried the JS methods by removing the duplication and made them more
  generic so the script can be re-used.
- Move the control of show/hide tabs to css, by using an `.active` class
- And move the js click event to be called from the javascript file
- To make that happen we need a `.sub-header .button` with a `data-target` attribute.
  We also need a `.tabs` area with nested `.tab `,
  and a class that matches the `data-target`
- What the JS does is pretty much hide all the tabs with a class of `.tab`
  and then only shows the one with the right class name
- Removed the feature specs because if JS is disabled,
  the user will still be able to see the families groupings.
- Removed the `sub_header` partial because it was only used in one place
  and it was simpler to keep the markup in the same view

Commit:
Clean up how tabs are being displayed

- Dried the JS methods by removing the duplication and made them more
  generic so the script can be re-used.
- Move the control of show/hide tabs to css, by using an `.active` class
- And move the js click event to be called from the javascript file
- To make that happen we need a `.sub-header .button` with a `data-target` attribute.
  We also need a `.tabs` area with nested `.tab `,
  and a class that matches the `data-target`
- What the JS does is pretty much hide all the tabs with a class of `.tab`
  and then only shows the one with the right class name
- Removed the feature specs because if JS is disabled,
  the user will still be able to see the families groupings.
- Removed the `sub_header` partial because it was only used in one place
  and it was simpler to keep the markup in the same view

Commit:
Clean up how tabs are being displayed

- Dried the JS methods by removing the duplication and made them more
  generic so the script can be re-used.
- Move the control of show/hide tabs to css, by using an `.active` class
- And move the js click event to be called from the javascript file
- To make that happen we need a `.sub-header .button` with a `data-target` attribute.
  We also need a `.tabs` area with nested `.tab `,
  and a class that matches the `data-target`
- What the JS does is pretty much hide all the tabs with a class of `.tab`
  and then only shows the one with the right class name
- Removed the feature specs because if JS is disabled,
  the user will still be able to see the families groupings.
- Removed the `sub_header` partial because it was only used in one place
  and it was simpler to keep the markup in the same view

Commit:
Clean up how tabs are being displayed

- Dried the JS methods by removing the duplication and made them more
  generic so the script can be re-used.
- Move the control of show/hide tabs to css, by using an `.active` class
- And move the js click event to be called from the javascript file
- To make that happen we need a `.sub-header .button` with a `data-target` attribute.
  We also need a `.tabs` area with nested `.tab `,
  and a class that matches the `data-target`
- What the JS does is pretty much hide all the tabs with a class of `.tab`
  and then only shows the one with the right class name
- Removed the feature specs because if JS is disabled,
  the user will still be able to see the families groupings.
- Removed the `sub_header` partial because it was only used in one place
  and it was simpler to keep the markup in the same view

Commit:
Change buttons to be links, and clean their styling

- Semantically, we click either links or actual form buttons
  so, changed the markup to be more semantically correct.
- Thus added `e.preventDefault();`
  so that the JS will not follow the link through.
- Also now adding the `.active` class to the buttons as well,
  to indicate which one is currently active.
- Cleaned up the styles to use sass, and incorporate the changes in
  markup. Also added a style for hovering the links.

Suggested TODO:
What would be good for the UX, is to actually
have the buttons to link to actual views.
So if the user has JS disabled, they can still see each group.

Suggested consideration:
Start thinking about how to handle when there are too many families.
One possible solution can be pagination.

Commit:
Change buttons to be links, and clean their styling

- Semantically, we click either links or actual form buttons
  so, changed the markup to be more semantically correct.
- Thus added `e.preventDefault();`
  so that the JS will not follow the link through.
- Also now adding the `.active` class to the buttons as well,
  to indicate which one is currently active.
- Cleaned up the styles to use sass, and incorporate the changes in
  markup. Also added a style for hovering the links.

Suggested TODO:
What would be good for the UX, is to actually
have the buttons to link to actual views.
So if the user has JS disabled, they can still see each group.

Suggested consideration:
Start thinking about how to handle when there are too many families.
One possible solution can be pagination.

Commit:
Change buttons to be links, and clean their styling

- Semantically, we click either links or actual form buttons
  so, changed the markup to be more semantically correct.
- Thus added `e.preventDefault();`
  so that the JS will not follow the link through.
- Also now adding the `.active` class to the buttons as well,
  to indicate which one is currently active.
- Cleaned up the styles to use sass, and incorporate the changes in
  markup. Also added a style for hovering the links.

Suggested TODO:
What would be good for the UX, is to actually
have the buttons to link to actual views.
So if the user has JS disabled, they can still see each group.

Suggested consideration:
Start thinking about how to handle when there are too many families.
One possible solution can be pagination.

Commit:
Change buttons to be links, and clean their styling

- Semantically, we click either links or actual form buttons
  so, changed the markup to be more semantically correct.
- Thus added `e.preventDefault();`
  so that the JS will not follow the link through.
- Also now adding the `.active` class to the buttons as well,
  to indicate which one is currently active.
- Cleaned up the styles to use sass, and incorporate the changes in
  markup. Also added a style for hovering the links.

Suggested TODO:
What would be good for the UX, is to actually
have the buttons to link to actual views.
So if the user has JS disabled, they can still see each group.

Suggested consideration:
Start thinking about how to handle when there are too many families.
One possible solution can be pagination.

Commit:
Add all_true_families and all_false_families traits into factories and use it in the feature specs

Commit:
Add all_true_families and all_false_families traits into factories and use it in the feature specs

Commit:
Add all_true_families and all_false_families traits into factories and use it in the feature specs

Commit:
Add all_true_families and all_false_families traits into factories and use it in the feature specs

scara82 reviewed (commented):
Thanks @elle. So do you suggest to add 3 different views for all/active/inactive families? I will do this and the pagination in the next PR. Thanks, it looks much better (also the button colour depending on which one is 'active')!

Commit:
deleted unsed files

Commit:
deleted unsed files

Commit:
deleted unsed files

Commit:
deleted unsed files

Commit:
Fix rebase leftovers

Commit:
Fix rebase leftovers

Commit:
Add ActiveFamiliesController with index only and move sub_header into a partial

Commit:
Add ActiveFamiliesController with index only and move sub_header into a partial

Commit:
Add AllFamiliesController with views

Commit:
Modify Families controller and view to use the All/Active/Inactive Controllers

Commit:
Modify Families controller and view to use the All/Active/Inactive Controllers

Commit:
Add InactiveFamiliesController

elle commented on app/views/families/index.html.erb:
```diff
@@ -29,15 +29,15 @@
 
   <div class="tab inactive-families">
-    <p class="title"><%= t(".not_active_families") %></p>
+    <p class="title"><%= t("families.index.inactive_families") %></p>
```
I believe you can use relative loading of a translation here: i.e. `t(".inactive_families")`

elle commented on app/views/families/index.html.erb:
```diff
@@ -20,7 +20,7 @@
 <section class="tabs">
   <div class="tab active-families active">
-    <p class="title"><%= t(".active_families") %></p>
+    <p class="title"><%= t("families.index.active_families") %></p>
```
Same ^

elle commented on app/views/families/index.html.erb:
```diff
@@ -29,15 +29,15 @@
 
   <div class="tab all-families">
-    <p class="title"><%= t(".all_families") %></p>
+    <p class="title"><%= t("families.index.all_families") %></p>
```
And here :)

elle commented on config/locales/en.yml:
```diff
@@ -54,9 +54,9 @@ en:
-        all: "All"
+        active: "ACTIVE"
+        inactive: "INACTIVE"
+        all: "ALL"
```
No need to make these all caps here, when we can define that as a stylesheet rule.

elle commented on app/assets/stylesheets/components/_tabs.scss:
```diff
@@ -0,0 +1,7 @@
+  display: none;
+}
+
+.active {
```
We can also define these:
```
.tab {
  display: none;

  &.active {
    display: block;
  }
}
```

Which will scope the `.active` behaviour to `.tab` elements only.
Question is: do we want to keep the `.active` more generic and use it elsewhere? or do we prefer the scoping?

scara82 replied:
Yes, I think it's better to keep it more generic, I'm pretty sure I can use it also for trips :)

scara82 replied:
I'm gonna to use the new JS for the trips in the next PR.

Commit:
Create folder for All/Active/InactiveFamilies Controllers feature specs and move the test inside the proper folder

Commit:
Create folder for All/Active/InactiveFamilies Controllers feature specs and move the test inside the proper folder

Commit:
Add translations for new Controllers titles

Commit:
Add translations for new Controllers titles

elle commented on spec/features/families/admin_views_all_active_families_spec.rb:
```diff
@@ -0,0 +1,15 @@
+require "rails_helper"
+
+feature "Admin can see all the active families when in families#index" do
```
A suggestion to change the description to be: "Admin sees active and inactive families"

Commit:
Remove the route get 'all_families/index' autogenerated

Commit:
Remove the route get 'all_families/index' autogenerated

elle commented on spec/features/families/admin_views_all_active_families_spec.rb:
```diff
@@ -0,0 +1,15 @@
+
+    expect(page).to have_text(I18n.t("families.index.buttons.active"))
+    expect(page).to have_text(true)
+    expect(page).not_to have_text(false)
```
A suggestion to change the feature spec to be more specific as to what we see on the page: i.e.
```
active_family = create(:all_true_family, last_name: "Russo")
inactive_family = create(:all_false_family, last_name: "De Luca")

visit families_path
 
expect(page).to have_text("Russo")
expect(page).to have_text("De Luca")
```

elle commented on spec/features/families/admin_views_all_families_spec.rb:
```diff
@@ -0,0 +1,16 @@
+
+      expect(page).to have_text(I18n.t("families.index.buttons.all"))
+      expect(page).to have_text("false")
+      expect(page).to have_text("true")
```
A suggestion to change this spec to be:
```
feature "Admin sees all active and inactive families" do
  describe "when click on the 'All' button" do
    scenario "successfully", js: true do
      active_family = create(:all_true_family, last_name: "Russo")
      inactive_family = create(:all_false_family, last_name: "De Luca")

      visit families_path

      expect(page).to have_text("Russo")
      expect(page).to have_text("De Luca")
    end
  end
end
```

scara82 replied:
Done in sha `f995a5f`

elle commented on app/models/family.rb:
```diff
@@ -1,3 +1,11 @@
   validates_presence_of :last_name, :capability
+
+  def self.active
+    where(active: true)
```
Missing spec for this scope

scara82 replied:
Done in sha: `4c2531c`

elle commented on app/models/family.rb:
```diff
@@ -1,3 +1,11 @@
+  end
+
+  def self.inactive
+    where(active: false)
```
And here: missing spec for this scope

elle replied:
Lines 4-5 can be removed

elle commented on app/views/families/index.html.erb:
```diff
@@ -1,6 +1,46 @@
+  <ul>
+    <li>
+      <a href="#" class="button active" data-target="active-families">
+        <%= t("families.index.buttons.active") %>
```
Can probably use relative find for translations here as well

elle commented on app/views/families/index.html.erb:
```diff
@@ -1,6 +1,46 @@
+    </li>
+    <li>
+      <a href="#" class="button" data-target="inactive-families">
+        <%= t("families.index.buttons.inactive") %>
```
And here ^

elle commented on app/views/families/index.html.erb:
```diff
@@ -1,6 +1,46 @@
+    </li>
+    <li>
+      <a href="#" class="button" data-target="all-families">
+        <%= t("families.index.buttons.all") %>
```
And here ^ :)

elle commented on app/views/families/index.html.erb:
```diff
@@ -1,6 +1,46 @@
+
+<section class="tabs">
+  <div class="tab active-families active">
+    <p class="title"><%= t("families.index.active_families") %></p>
```
One more ^

elle commented on app/views/families/index.html.erb:
```diff
@@ -1,6 +1,46 @@
+  </div>
+
+  <div class="tab inactive-families">
+    <p class="title"><%= t("families.index.inactive_families") %></p>
```
Relative find for translations ^

elle commented on app/views/families/index.html.erb:
```diff
@@ -1,6 +1,46 @@
+  </div>
+
+  <div class="tab all-families">
+    <p class="title"><%= t("families.index.all_families") %></p>
```
Last one ^

elle commented on app/assets/stylesheets/patterns/_links.scss:
```diff
@@ -1,4 +1,5 @@
 a {
   color: black;
   text-decoration: none;
+  float: left;
```
Alphabetical order?

scara82 replied:
Done in sha `e4d45f4`

elle commented on app/controllers/all_families_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class AllFamiliesController < ApplicationController
+  def index
+    @families = Family.all
```
A suggestion to implement pagination here as well. Maybe in another PR?

scara82 replied:
I was thinking to add to the other PR once this is merged :)

elle commented on app/controllers/application_controller.rb:
```diff
@@ -9,5 +9,10 @@ class ApplicationController < ActionController::Base
+  end
+
+  def default_url_options(options = {})
+    { locale: I18n.locale }.merge(options)
```
👍

elle commented on app/models/family.rb:
```diff
@@ -1,3 +1,13 @@
+    where(active: false)
+  end
+  scope :active, -> { where("active = true") }
+  scope :inactive, -> { where("active = false") }
```
These two lines are not needed. They repeat the above two methods.

scara82 replied:
Done in sha: `4c2531c`

elle commented on app/views/application/_language.html.erb:
```diff
@@ -0,0 +1,5 @@
+  <%= link_to "It", { locale: "it" } %>
+<% else %>
+  <%= link_to "En", { locale: "en" } %>
+<%end%>
```
Something to consider: always show all the available languages? so then the `if` block is not needed

scara82 replied:
Done in sha `14928cf`

elle commented on app/views/families/_form.html.erb:
```diff
@@ -1,14 +1,45 @@
-    <%= f.submit t(".submit") %>
+  <ul>
+    <li>
+      <%= f.label Family.human_attribute_name(:first_name) %>
```
A question: why do we need to explicitly define the labels? Does simple form not pick up the right translations?

scara82 replied:
Yes but in this way it is easy to style it.. :) I was having difficulties to position labels and inputs.

elle commented on app/views/families/_sub_header.html.erb:
```diff
@@ -0,0 +1,19 @@
+<nav class="sub-header">
+  <ul>
+    <li>
+      <a href="active_families" class="button" data-target="active-families">
```
`href` here should link to the right index view. A suggestion to also use a `link_to` helper... i.e.:
```
link_to t("families.index.buttons.active"), active_families_path, class: "button", data: { target: "active-families" }
```

scara82 replied:
Done in sha `6b83801`

elle commented on app/views/families/_sub_header.html.erb:
```diff
@@ -0,0 +1,19 @@
+    <li>
+      <a href="inactive_families" class="button" data-target="inactive-families">
+        <%= t("families.index.buttons.inactive") %>
+      </a>
```
Same here ^

scara82 replied:
Done in sha 6b83801

elle commented on app/views/families/_sub_header.html.erb:
```diff
@@ -0,0 +1,19 @@
+    </li>
+    <li>
+      <a href="all_families" class="button" data-target="all-families">
+        <%= t("families.index.buttons.all") %>
```
And here ^

scara82 replied:
Done in sha 6b83801

elle commented on app/views/pages/home.html.erb:
```diff
@@ -1,18 +1,18 @@
   <tr>
     <td>
-      <%= link_to t("families.new.title"), new_family_path %>
+      <%= link_to t("families.new.title"), new_family_path(locale: I18n.locale) %>
```
Do we still need to add the locale to the paths after defining the default link params?

scara82 replied:
Ops, I missed it! Thanks! Done in sha `455ebcb`

elle commented on app/views/schools/_form.html.erb:
```diff
@@ -1,9 +1,25 @@
-    <%= f.submit t(".submit")%>
+  <ul>
+    <li>
+      <%= f.label School.human_attribute_name(:name) %>
```
Same question here about simple form...

elle commented on app/views/trips/_form.html.erb:
```diff
@@ -1,13 +1,37 @@
-    <%= f.submit t(".submit") %>
+  <ul>
+    <li>
+      <%= f.label Trip.human_attribute_name(:total_girls) %>
```
Still the same question...

elle commented on config/application.rb:
```diff
@@ -17,6 +17,9 @@ module Domus
   class Application < Rails::Application
     config.app_generators.scaffold_controller :responders_controller
 
+    config.i18n.load_path += Dir[Rails.root.join("config", "locales", "**", "*.{rb,yml}")]
```
A question: why do we need to define this `load_path`? Isn't it loaded by default?

elle commented on spec/controllers/active_families_controller_spec.rb:
```diff
@@ -0,0 +1,10 @@
+  describe "GET #index" do
+    it "returns http success" do
+      get :index
+      expect(response).to have_http_status(:success)
```
No real need for this spec. I would change that to be a feature spec.

scara82 replied:
Deleted in sha 4f22698

elle commented on spec/controllers/all_families_controller_spec.rb:
```diff
@@ -0,0 +1,10 @@
+  describe "GET #index" do
+    it "returns http success" do
+      get :index
+      expect(response).to have_http_status(:success)
```
Same here. Suggesting to change this to be a feature spec.

scara82 replied:
Deleted in sha 4f22698

elle commented on spec/controllers/inactive_families_controller_spec.rb:
```diff
@@ -0,0 +1,10 @@
+  describe "GET #index" do
+    it "returns http success" do
+      get :index
+      expect(response).to have_http_status(:success)
```
One last one :)

scara82 replied:
Deleted in sha 4f22698

elle commented on spec/features/all_families/admin_views_all_families_spec.rb:
```diff
@@ -0,0 +1,16 @@
+      create(:all_false_family)
+
+      visit families_path
+      click_link("All")
```
Translation

scara82 replied:
Done in sha `f995a5f`

elle commented on spec/features/active_families/admin_views_all_active_families_spec.rb:
```diff
@@ -0,0 +1,14 @@
+    create(:all_false_family)
+
+    visit families_path
+    click_link("Active")
```
Translation

scara82 replied:
Done in sha `f995a5f`

elle commented on spec/features/families/admin_views_all_inactive_families_spec.rb:
```diff
@@ -0,0 +1,17 @@
+require "rails_helper"
+
+feature "Admin can see all the not active families when in families#index" do
```
A suggestion to change the description: "Admin sees all the inactive families"

scara82 replied:
Done in sha `f995a5f`

elle commented on spec/features/families/admin_views_all_inactive_families_spec.rb:
```diff
@@ -0,0 +1,17 @@
+require "rails_helper"
+
+feature "Admin can see all the not active families when in families#index" do
+  context "when click on the 'Not Active' button" do
```
Change to `'Inactive'`

scara82 replied:
Done in sha `f995a5f`

elle commented on spec/features/families/admin_views_all_inactive_families_spec.rb:
```diff
@@ -0,0 +1,17 @@
+
+feature "Admin can see all the not active families when in families#index" do
+  context "when click on the 'Not Active' button" do
+    scenario "successfully", js: true do
```
Since we made the views work without JS, we no longer need the `js: true` here

scara82 replied:
Done in sha `f995a5f`

elle commented on spec/features/inactive_families/admin_views_all_inactive_families_spec.rb:
```diff
@@ -0,0 +1,16 @@
+
+feature "Admin can see all the not active families when in families#index" do
+  context "when click on the 'Not Active' button" do
+    scenario "successfully", js: true do
```
Same here ^

scara82 replied:
Done in sha `f995a5f`

elle commented on spec/features/inactive_families/admin_views_all_inactive_families_spec.rb:
```diff
@@ -0,0 +1,16 @@
+      create(:all_false_family)
+
+      visit families_path
+      click_link("Inactive")
```
A suggestion to be consistent and use the `id` attribute here like in the other spec

scara82 replied:
I changed all the specs to have the title before reading your comment, I hope it is ok. :)

Commit:
Delete specs files moved into their folders

Commit:
Alphabetical order

Commit:
Add active and inactive scope specs

Commit:
Remove if statement to display both links to language in the header

Commit:
Converted href into link_to for families/sub_header

Commit:
Remove the locale that is no more required

Commit:
Fixes the features specs: remove js: true, reword it,add translations

Commit:
Fixes mistakes from rebasing

elle commented on app/views/application/_language.html.erb:
```diff
@@ -1,5 +1,2 @@
-  <%= link_to "En", { locale: "en" } %>
-<%end%>
+<%= link_to "It", { locale: "it" } %>
+<%= link_to "En", { locale: "en" } %>
```
Use translations

scara82 replied:
Done in sha `776504b`

elle commented on app/views/families/_form.html.erb:
```diff
@@ -35,7 +35,7 @@
     <li>
       <%= f.label Family.human_attribute_name(:active) %>
-      <%= f.input_field :active, as: :radio_buttons, collection: ["Y", "N"] %>
+      <%= f.input_field :active, as: :radio_buttons, collection: ["true", "false"] %>
```
I think that we can remove the explicit collection here

scara82 replied:
Done in sha `8b313ef`

elle commented on app/views/families/_sub_header.html.erb:
```diff
@@ -0,0 +1,16 @@
+  <ul>
+    <li>
+      <%= link_to t("families.index.buttons.active"), active_families_path,
+        class: "button", data: { target: "active-families" } %>
```
👍

elle commented on app/views/families/index.html.erb:
```diff
@@ -1,6 +1,28 @@
-  <%= render "families_table_header" %>
-  <%= render @families %>
-</table>
+<%= render "sub_header"%>
```
Since this sub_header is only used in one place, a suggestion to roll the markup back into this view

scara82 replied:
Done in sha `88c5afe`

elle commented on app/views/pages/home.html.erb:
```diff
@@ -2,17 +2,17 @@
   <tr>
     <td>
-      <%= link_to t("schools.new.title"), new_school_path(locale: I18n.locale) %>
+      <%= link_to t("schools.new.title") %>
```
You still need the link, just don't need to pass the locale :)

scara82 replied:
I can't understand....is not correct?  🤔

elle commented on spec/features/all_families/admin_views_all_families_spec.rb:
```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+feature "Admin can see all the families when in families#index" do
```
A suggestion to change the title to be more accurate: "Admin sees all the families"

scara82 replied:
Done in sha: `f92ad52`

elle commented on spec/features/families/admin_views_all_families_spec.rb:
```diff
@@ -0,0 +1,15 @@
+require "rails_helper"
+
+feature "Admin sees all active and inactive families" do
+  describe "when click on the 'All' button" do
```
Line 4 can be removed because it is no longer implemented in this feature spec

scara82 replied:
Done in sha: `f92ad52`

elle commented on spec/features/inactive_families/admin_views_all_inactive_families_spec.rb:
```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+feature "Admin can see all the inactive families when in families#index" do
```
Again, suggesting to rename to something more accurate: "Admin sees inactive families only"

scara82 replied:
Done in sha: `f92ad52`

elle commented on spec/models/family_spec.rb:
```diff
@@ -5,4 +5,18 @@
     it { is_expected.to validate_presence_of(:capability) }
   end
+
+  describe "scope" do
```
I would change this block to be:
```
describe ".active" do
  it...
end

describe ".inactive" do
  it ...
end
```

scara82 replied:
Done previously (I'm sorry I can't find the sha :) )

scara82 commented:
Closing this PR in favour of https://github.com/hooroo/domus/pull/36
